### DISCUSSION TITLE
Invalid value used as weak map key

### DISCUSSION THREAD

**Discussion by husayt** [vuejs/pinia#460] [Help and Questions] [answered]
Hi, 
I get this error when doing a build, all works fine in dev. Happens after I added a composition method which uses a pinia store.

Not sure if this is an error on my side, or a bug with Pinia.

![image](https://user-images.githubusercontent.com/1169324/116387315-9bb2dc00-a812-11eb-972f-c421a7574c50.png)

Composition function looks like this
```js
function useCurFetcher({}) {
 \\ this causes build exception
  const nav = useNavStore()
  ....
  }

export const curFetcher = useCurFetcher({})
```
Setup: 
* Vue 3
* pinia@2.0.0-alpha.13
* vite 2
* Vite-SSG

**Comment by posva** [comment, marked as answer]
I can't tell just from that but if `useCurFetch()` is called outside of `setup()` (cf docs), it will def error unless you pass pinia to useNaveStore

**Reply by husayt** [reply]
I have changed the call point and that worked. Thanks

**Reply by husayt** [reply]
Sumbled on this again. Need to make sure to wrap for SSR
```
 if (!import.meta.env.SSR) {
    const strg = useMyStore()
...
}
```
